cli/cmd/plugin/ui/web/tanzu-ui/src/components/aws/credential-step/Credentials.tsx

```diff
@@ -0,0 +1,272 @@
+    });
+    const {
+        register,
+        handleSubmit,
+        setValue,
+        formState: { errors },
+    } = methods;
+
+    const [type, setType] = useState<CREDENTIAL_TYPE>(CREDENTIAL_TYPE.PROFILE);
+    const [connected, setConnection] = useState(false);
+
+    const [regions, setRegions] = useState<string[]>([]);
+    const [keypairs, setKeyPairs] = useState<AWSKeyPair[]>([]);
+
+    useEffect(() => {
+        // fetch regions
+        AwsService.getAwsRegions().then((data) => setRegions(data));
+    }, []);
+
+    useEffect(() => {
+        if (connected) {
+            AwsService.getAwsKeyPairs().then((data) => {
+                setKeyPairs(data);
+            });
+        }
+    }, [connected]);
+
+    const selectCredentialType = (event: ChangeEvent<HTMLSelectElement>) => {
+        setConnection(false);
+        setType(CREDENTIAL_TYPE[event.target.value as CREDENTIAL_TYPE]);
+    };
+
+    const onSubmit: SubmitHandler<FormInputs> = (data) => {
+        if (connected && Object.keys(errors).length === 0) {
+            if (goToStep && currentStep && submitForm) {
+                goToStep(currentStep + 1);
+                submitForm(currentStep);
+            }
+        }
+    };
+
+    const handleConnect = () => {
+        let params: AWSAccountParams = {};
+        if (type === CREDENTIAL_TYPE.PROFILE) {
+            params = {
+                profileName: awsState.data.PROFILE,
+                region: awsState.data.REGION,
+            };
+        } else {
+            params = {
+                accessKeyID: '',
+                region: awsState.data.REGION,
+                secretAccessKey: '',
+                sessionToken: '',
+            };
+        }
+        AwsService.setAwsEndpoint(params).then(() => {
+            setConnection(true);
+        });
+    };
+
+    const handleSelectProfile = (profile: string) => {
+        setConnection(false);
+        if (handleValueChange) {
+            setTimeout(() => {
+                handleValueChange('PROFILE', profile, currentStep, errors);
+            });
+        }
+    };
+
+    const handleSelectRegion = (region: string) => {
+        setConnection(false);
+        if (handleValueChange) {
+            setTimeout(() => {
+                handleValueChange('REGION', region, currentStep, errors);
+            });
+        }
+    };
+
+    const handleSelectKeyPair = (event: ChangeEvent<HTMLSelectElement>) => {
+        setValue('EC2_KEY_PAIR', event.target.value, { shouldValidate: true });
+        if (handleValueChange) {
+            setTimeout(() => {
+                handleValueChange(
+                    'EC2_KEY_PAIR',
+                    event.target.value,
+                    currentStep,
+                    errors
+                );
+            });
+        }
+    };
+
+    const handleInputChange = (field: string, value: string) => {
+        setConnection(false);
+        if (handleValueChange) {
+            // setTimeout(() => {
```

Comment:
remove unused setTimeout

Reply:
I removed //. this is a workaround and will find a better solution later.